mtags/src/main/scala/scala/meta/internal/pc/TypeDefinitionProvider.scala

```diff
@@ -0,0 +1,80 @@
+package scala.meta.internal.pc
+
+import java.nio.file.Files
+import org.eclipse.{lsp4j => l}
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.pc.OffsetParams
+import scala.meta.internal.jdk.CollectionConverters._
+import scala.meta.internal.metals.CompilerOffsetParams
+
+class TypeDefinitionProvider(val compiler: MetalsGlobal, params: OffsetParams)
+    extends ProviderCommon {
+  import compiler._
+
+  def typedTree: Option[Tree] = {
+    if (params.isWhitespace) {
+      None
+    } else {
+      val unit = addCompilationUnit(
+        code = params.text(),
+        filename = params.filename(),
+        cursor = None
+      )
+      val pos = unit.position(params.offset())
+      val tree = typedHoverTreeAt(pos)
+      Some(tree.asInstanceOf[compiler.Tree])
+    }
+  }
+
+  def typeSymbol: Option[Symbol] = {
+    val t = typedTree
+    t.flatMap {
+      case tree
+          if tree.symbol.isTypeSymbol || tree.symbol.isMethod || tree.symbol.isConstructor =>
+        Some(tree.symbol)
+      case tree if tree.tpe.isDefined =>
+        Some(tree.tpe.typeSymbol)
+      case tree if tree.children.nonEmpty =>
+        Some(tree.children.head.tpe.typeSymbol)
+      case tree =>
+        val expTree = expandRangeToEnclosingApply(tree.pos)
+        if (expTree.tpe.isDefined) Some(expTree.tpe.typeSymbol)
+        else None
+      case _ => None
+    }
+  }
+
+  def typeDefinition: List[l.Location] = {
+    typeSymbol.map {
+      case sym
+          if sym.pos.source.file != null && !sym.pos.source.file.isVirtual =>
+        val pos = sym.pos
+
+        val src = Files.readAllLines(sym.pos.source.file.file.toPath).asScala
+
+        val nParams = CompilerOffsetParams(
+          pos.source.path,
+          src.mkString,
+          pos.start
+        )
+
+        val provider = new PcDefinitionProvider(compiler, nParams)
+        val res = provider.definition()
```

Comment:
`res` is a redundant variable, let's rename it to `val locations = provider.definition().locations()`

Reply:
Fixed.